src/brogue/Recordings.c

```diff
@@ -726,6 +726,205 @@ void pausePlayback() {
+                message("Saved.", true);
+                rogue.gameHasEnded = true;
+            } else {
+                askAgain = true;
+            }
+        }
+    } while (askAgain);
+    deleteMessages();
+}
+
+void saveRecordingNoPrompt(char *filePath)
+{
+    if (rogue.playbackMode) {
+        return;
+    }
+    getAvailableFilePath(filePath, "Recording", RECORDING_SUFFIX);
+    strcat(filePath, RECORDING_SUFFIX);
+    remove(filePath);
+    rename(currentFilePath, filePath);
+}
+
+void saveRecording(char *filePath) {
+    char defaultPath[BROGUE_FILENAME_MAX];
+    boolean askAgain;
+
+    if (rogue.playbackMode) {
+        return;
+    }
+
+    getAvailableFilePath(defaultPath, "Recording", RECORDING_SUFFIX);
+
+    deleteMessages();
+    do {
+        askAgain = false;
+        if (getInputTextString(filePath, "Save recording as (<esc> to cancel): ",
+                               BROGUE_FILENAME_MAX - strlen(RECORDING_SUFFIX), defaultPath, RECORDING_SUFFIX, TEXT_INPUT_FILENAME, false)) {
+
+            strcat(filePath, RECORDING_SUFFIX);
+            if (!fileExists(filePath) || confirm("File of that name already exists. Overwrite?", true)) {
+                remove(filePath);
+                rename(currentFilePath, filePath);
+            } else {
+                askAgain = true;
+            }
+        } else { // Declined to save recording; save it anyway as LastRecording, and delete LastRecording if it already exists.
+            strcpy(filePath, LAST_RECORDING_NAME);
+            strcat(filePath, RECORDING_SUFFIX);
+            if (fileExists(filePath)) {
+                remove(filePath);
+            }
+            rename(currentFilePath, filePath);
+        }
+    } while (askAgain);
+    deleteMessages();
+}
+
+void copyFile(char *fromFilePath, char *toFilePath, unsigned long fromFileLength) {
+    unsigned long m, n;
+    unsigned char fileBuffer[INPUT_RECORD_BUFFER];
+    FILE *fromFile, *toFile;
+
+    remove(toFilePath);
+
+    fromFile    = fopen(fromFilePath, "rb");
+    toFile      = fopen(toFilePath, "wb");
+
+    for (n = 0; n < fromFileLength; n += m) {
+        m = min(INPUT_RECORD_BUFFER, fromFileLength - n);
+        fread((void *) fileBuffer, 1, m, fromFile);
+        fwrite((void *) fileBuffer, 1, m, toFile);
+    }
+
+    fclose(fromFile);
+    fclose(toFile);
+}
+
+// at the end of loading a saved game, this function transitions into active play mode.
+void switchToPlaying() {
+    char lastGamePath[BROGUE_FILENAME_MAX];
+
+    getAvailableFilePath(lastGamePath, LAST_GAME_NAME, GAME_SUFFIX);
+    strcat(lastGamePath, GAME_SUFFIX);
+
+    rogue.playbackMode          = false;
+    rogue.playbackFastForward   = false;
+    rogue.playbackOmniscience   = false;
+    locationInRecordingBuffer   = 0;
+    copyFile(currentFilePath, lastGamePath, recordingLocation);
+
+#ifndef ENABLE_PLAYBACK_SWITCH
+    if(DELETE_SAVE_FILE_AFTER_LOADING) {
+      remove(currentFilePath);
+    }
+#endif
```

Comment:
I presume this is to stop the recording file being deleted when we switch. But I think this also stops save files being deleted after loading? I wonder if there's a way to distinguish each behaviour

Reply:
When save files are loaded, switchToPlaying is called from loadSavedGame. But from a recording, it comes from executePlaybackInput. Those are the only two places it is called, so it can definitely be distinguished

Reply:
Ok, this behaviour is ok for now, will explore a change myself